docs/resources/ocean_automatic_rightsizing.md

```diff
@@ -0,0 +1,195 @@
+---
+layout: "spotinst"
+page_title: "Spotinst: ocean_right_sizing_rule"
+subcategory: "Ocean"
+description: |-
+  Provides a Spotinst Ocean Right Sizing resource.
+---
+
+# spotinst\_ocean\_right\_sizing\_rule
+
+Manages a Spotinst Ocean right sizing rule resource.
+
+## Example Usage
+
+```hcl
+resource "spotinst_ocean_right_sizing_rule" "example" {
+  ocean_id = "o-123456"
+  rule_name = "test-rule"
+  exclude_preliminary_recommendations= true
+  restart_replicas="ALL_MANIFEST"
+
+  recommendation_application_hpa{
+    allow_hpa_recommendation= true
+  }
+
+  recommendation_application_intervals {
+    repetition_basis = "WEEKLY"
+    weekly_repetition_basis {
+      interval_days = ["MONDAY", "WEDNESDAY"]
+      interval_hours_start_time = "12:00"
+      interval_hours_end_time = "14:00"
+    }
+  }
+
+
+  recommendation_application_intervals {
+    repetition_basis = "MONTHLY"
+    monthly_repetition_basis {
+      interval_months = [2,6,9]
+      week_of_the_month = ["FIRST","LAST"]
+      weekly_repetition_basis {
+        interval_days = ["MONDAY"]
+        interval_hours_start_time = "03:00"
+        interval_hours_end_time = "04:00"
+      }
+    }
+  }
+
+  recommendation_application_boundaries {
+    cpu_min = 120
+    cpu_max = 190
+    memory_min = 160
+    memory_max = 190
+  }
+
+  recommendation_application_min_threshold {
+    cpu_percentage = 0.412
+    memory_percentage = 0.36
+  }
+
+  recommendation_application_overhead_values {
+    cpu_percentage = 0.80
+    memory_percentage = 0.50
+  }
+  
+}
+```
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `ocean_id` - (Required) Identifier of the Ocean cluster.
+* `rule_name` - (Required) The unique name of the rule.
+* `exclude_preliminary_recommendations` - (Optional) Exclude preliminary recommendations (recommendations based on less than 4 full days of data).
+* `restart_replicas` - (Optional) Enum: "MORE_THAN_ONE_REPLICA" "ALL_MANIFEST" "NO_RESTART"
+  Enable to sequentially restart pod batches according to recommendations, for all pods, only more than 1 replica, or not any pod.
+* `recommendation_application_boundaries` - (Optional) Describes the Ocean Kubernetes Auto Scaler.Determines the Ocean Rightsizing rule recommendation application boundaries.
```

Comment:
please correct the description according to openApi document

Reply:
Updated